.travis.yml

```diff
@@ -135,6 +135,25 @@ script:
       python -m iris.tests.runner --example-tests --print-failed-images;
     fi
 
+  # A call to check "whatsnew" contributions are valid, because the Iris test
+  # for it needs a *developer* install to be able to find the docs.
+  - if [[ ${TEST_TARGET} == 'doctest' ]]; then
+      cd ${INSTALL_DIR}/docs/iris/src/whatsnew;
+      python aggregate_directory.py --checkonly;
+    fi
+
+  # When pushing built docs, attempt to make a preliminary whatsnew with 'aggregate_directory.py', before the build.
+  - >
+    if [[ ${PUSH_BUILT_DOCS} == 'true' ]]; then
+      cd ${INSTALL_DIR}/docs/iris/src/whatsnew;
+      WHATSNEW=$(ls -d contributions_* 2>/dev/null);
+      if [[ "$WHATSNEW" != "" ]]; then
+        VERSION=$(echo $WHATSNEW | grep -o "[[:digit:].]*");
+        python aggregate_directory.py $VERSION;
```

Comment:
I think aggregate_directory should also take an optional flag to set the release date to be something like "Unreleased" or "TBC"
Currently, it [uses the current date](https://github.com/SciTools/iris/blob/7419f8839510d8b784bcca9567fe5ac9a9aaa68a/docs/iris/src/whatsnew/aggregate_directory.py#L180) so if you look at the [Iris 2.3 whats new](https://scitools-docs.github.io/iris/auto_branch_docs/whatsnew/2.3.html) that is generated, it claims that it has already been released...
I think this will just lead to confusion

Reply:
ok